src/backend/services/chat.py

```diff
@@ -174,6 +158,65 @@ def process_chat(
     )
 
 
+def check_chat_request_tools(
+    chat_request: BaseChatRequest, entity: Agent | User
+) -> None:
+    if chat_request.tools:
+        entity_type = "agent" if isinstance(entity, Agent) else "user"
```

Comment:
Best to explicitly validate in the else. scenario if `entity` is `User`, and throw some exception if its neither Agent|User

Reply:
Thanks

Reply:
Thanks